### PR TITLE
Add `Not available under this OS` error documentation to `faq.md` under General

### PR DESCRIPTION
**Pull request by salahdin-ahmed**
Hello,

Since I thought that this is a simple piece of documentation, I worked on it. It closes the comment [#282 (comment)](https://github.com/kmonad/kmonad/issues/282#issuecomment-863001153). I think this is just a good starting point. It should be rewritten or modified later though.

Thanks for the time.

### CODE REVIEW

File: doc/faq.md [hunk 1]
```diff
@@ -160,3 +160,16 @@ creating a numpad in the middle of the laptop keyboard. This remapping happens
 in the hardware, before any event is ever registered with the operating system,
 therefore KMonad has no way to 'get' at any of those events. This means that we
 cannot remap them in any way.
+
+### Q: When I run KMonad I get error `Not available under this OS`
+
+A: This error occurs when there is OS-specific options in the used configuration
```
Comment (zac-ng) [inline doc/faq.md:166]: This error occurs when there are OS-specific options in the used configuration

**Review by zac-ng — APPROVED** [review]
Good addition to faq, some small typos fixes suggested.  Feel free to disregard them though, since I'm not that familiar with the code review process.

File: doc/faq.md [hunk 2]
```diff
@@ -160,3 +160,16 @@ creating a numpad in the middle of the laptop keyboard. This remapping happens
...
+file.  Usually this happens when you are on windows and try running the tutorial
+file and do not comment, or even delete, the Linux options in `defcfg` and
```
[the remark this answers is not in the document]
Comment (slotThe) [reply doc/faq.md:168]: perhaps even "comment out", to be extra clear

File: doc/faq.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (salahdin-ahmed) [reply doc/faq.md]: Fixed in https://github.com/kmonad/kmonad/pull/285/commits/902a702def7967a4b4116d3ce401f6877043a739

File: doc/faq.md [hunk 3]
```diff
@@ -160,3 +160,16 @@ creating a numpad in the middle of the laptop keyboard. This remapping happens
...
+uncomment the Windows options. Nevertheless, this still can happen on other
+operating systems, the error message changes slightly based on the operating
+system (e.g. `Not available under this OS: LowLevelHookSource`, `Not available
+under this OS: DeviceSource`), but they all starts with `Not available under
```
[the remark this answers is not in the document]
Comment (salahdin-ahmed) [reply doc/faq.md]: Resolved in https://github.com/kmonad/kmonad/pull/285/commits/29a13654f862b46f0ac409c5e7411f4e9a29211a

File: doc/faq.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (salahdin-ahmed) [reply doc/faq.md]: Resolved in https://github.com/kmonad/kmonad/pull/285/commits/6599b1c946ee8a2e3c7d2162e2c538b572227bcf

**Comment by salahdin-ahmed** [comment]
Are there any other notes? If there is none, I will squash the commits!

File: doc/faq.md [hunk 4]
```diff
@@ -160,3 +160,16 @@ creating a numpad in the middle of the laptop keyboard. This remapping happens
 in the hardware, before any event is ever registered with the operating system,
 therefore KMonad has no way to 'get' at any of those events. This means that we
 cannot remap them in any way.
+
+### Q: When I run KMonad I get error `Not available under this OS`
+
+A: This error occurs when there are OS-specific options in the used configuration
+file.  Usually this happens when you are on windows and try running the tutorial
```
Comment (slotThe) [inline doc/faq.md:167]:
```suggestion
file. Usually this happens when you are on windows, try to run the tutorial
```

**Comment by slotThe** [comment]
Just a minor thing from my side.  After that feel free to squash!

**Comment by salahdin-ahmed** [comment]
The squashing is done, I think the branch is now ready to be merged.

**Comment by slotThe** [comment]
Thanks!

**Pull request merged**